docs/data/migration/migration-pickers-v5/migration-pickers-v5.md

```diff
@@ -759,35 +773,59 @@ For example, the `ToolbarComponent` has been replaced by a `Toolbar` component s
 
 - The component slot `LeftArrowButton` has been renamed to `PreviousIconButton`:
 
-```diff
- <DatePicker
-   components={{
--    LeftArrowButton: CustomButton,
-+    PreviousIconButton: CustomButton,
-   }}
+  ```diff
+   <DatePicker
+  -  components={{
+  -    LeftArrowButton: CustomButton,
+  -  }}
+  +  components={{
+  +    PreviousIconButton: CustomButton,
+  +  }}
+     // or using the new `slots` prop
+  +  slots={{
+  +    previousIconButton: CustomButton,
+  +  }}
```

Comment:
I do not agree on the it's more complicated. I think when writing docs, we tend to put minimal information.

But when you are doing the modification, and get a message saying "components props is deprecated". You're happy to have a solution in the docs diff, and not in n introduction paragraph you skipped two sections before.

---

By the way their is a concern about the codemods. They are made assuming `slots`/`slotsProps` are new props.

So if the developer is doing this modification choosing `slots`, and then run the codemod transforming `components` to `slots` they will get the following diff. The props objects are not merged.

```diff
-components={{ Toolbar: CustomToolbar }}
-slots={{ previousIconButton: CustomButton }}
+slots={{ toolbar: CustomToolbar }}
+slots={{ previousIconButton: CustomButton }}
```

Reply:
> I do not agree on the "it's more complicated". I think when writing docs, we tend to put minimal information, and put it at only one place. But for migration guide, I'm not sure it's the best approach since we don't have to keep it up to date, and nobody will get familiar with the page up to know where information is.
> 
> When you are doing the modification and get a message saying "components props is deprecated". You're happy to have a solution in the docs diff, and not in n introduction paragraph you skipped two sections before.

Those are some great points you've made.
Migration guide should indeed receive special treatment in making it as clear as possible.
If everyone else agrees with the suggested approach—I'm not gonna be the one saying no. 👍

Reply:
But in such a case, we have to add a warning for the codemod about `slots`/`slotProps` to say something like

> If you are in a mix between `components` and `slots` (especially due to previous migration steps) after running this codemod, you should check the diff to see if it created multiple `slots`/`slotProps` props. I such senario, you can merge them manually

Reply:
What do you guys think about pushing this PR for this week release as well? 🤔 
This misalignment is causing a few extra issues to pop-up. 🤔 
IMHO, we could introduce an `info` or `success` callout in the components->slots codemod section with a text that Alex has suggested and it does seem like a nice compromise.
What do you think @joserodolfofreitas?

Reply:
@LukasTy @alexfauquette concerning the warning, where do you want to put it ?
We are never talking about the slot codemod in the migration guide.

Reply:
We are talking about it at the end in [Rename components to slots (optional)](https://deploy-preview-8341--material-ui-x.netlify.app/x/migration/migration-pickers-v5/#rename-components-to-slots-optional)

Reply:
Warning added